### PR TITLE
[MIG] stock_available_sale: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Quotations in quantity available to promise
===========================================

This module computes the quoted quantity of each Product, and subtracts it from
the quantity available to promise.

"Quoted" is defined as the sum of the quantities of this Product in
Sale Quotations, taking the context's shop or warehouse into account.

cc @Tecnativa

### CODE REVIEW

**Review by gurneyalex — APPROVED** [review]

File: stock_available_sale/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        if to_date:
+            domain.extend([
+                ('order_id.requested_date', '<=', to_date),
+                '&',  # only consider 'date' when 'requested_date' is empty
+                ('order_id.requested_date', '=', False),
+                ('order_id.date', '<=', to_date),
+                ])
+        # Compute the quoted quantity for each product
+        results = Counter()
+        for group in self.env['sale.order.line'].read_group(
```
Comment (rousseldenis) [inline stock_available_sale/models/product_product.py]: @chienandalu As there is a performance problem in Odoo which maybe will not be solved https://github.com/odoo/odoo/issues/18679

It is recommended to execute read_group for computation purpose with `with_context(lang='')`

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

File: stock_available_sale/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+             "Quotation processed in this Warehouse.\n"
+             "In a context with a single Stock Location, this includes "
+             "Quotation processed at any Warehouse using "
+             "this Location, or any of its children, as it's Stock "
+             "Location.\n"
+             "Otherwise, this includes every Quotation.",
+    )
+
+    @api.depends('quoted_qty')
+    def _compute_immediately_usable_qty(self):
```
Comment (rousseldenis) [inline stock_available_sale/models/product_product.py:33]: @chienandalu Following codecoverage, it seems that this is not called in tests

**Comment by chienandalu** [comment]
@rousseldenis Changes done

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@Cedric-Pigeon

**Comment by pedrobaeza** [comment]
@chienandalu please squash before merging

**Comment by chienandalu** [comment]
@pedrobaeza done

**Pull request merged**